crates/sui-bridge/src/config.rs

```diff
@@ -417,6 +418,43 @@ pub fn read_bridge_client_key(path: &PathBuf) -> Result<SuiKeyPair, anyhow::Erro
         .map_err(|e| anyhow!("Error decoding authority key: {:?}", e))
 }
 
+/// Read a SuiKeyPair from a file. The content could be any of the following:
+/// - Base64 encoded `flag || privkey` for ECDSA key
+/// - Base64 encoded `privkey` for Raw key
+/// - Hex encoded `privkey` for Raw key
+/// If `require_secp256k1` is true, it will return an error if the key is not Secp256k1.
+pub fn read_key(path: &PathBuf, require_secp256k1: bool) -> Result<SuiKeyPair, anyhow::Error> {
```

Comment:
this model of trying on or the other feels a bit fragile. 
Is there no other way to do it?